### ISSUE TITLE
3.4-beta3: timing goes crazy after an iOS app suspend/resume cycle > 1 hour

### ISSUE THREAD

**Issue by oeleo1** [godotengine/godot#51482]
### Godot version

3.4-beta3

### System information

iOS, iPad Pro 1st gen (9.7 inch)

### Issue description

Resuming after 1 hour a suspended (exited) app on iOS does not resume the game normally. The timing is totally messed up and the game goes like crazy, including physics, particles, shaders using `TIME`. Everything runs at lightning (machine) speed.

I ran the same test on an Android phone (Samsung Galaxy A40 - mid low-range), on an iPad Pro 1st gen (old, slower) and on an iPhone 12 mini (recent, fast).

Interestingly only the iPad Pro 1st gen exhibits this behavior, the Android and the iPhone are OK. The iPhone and iPad run the same iOS version 14.7.1.

This is a regression since the same test on 3.3.2 works well on the iPad on resume after 1 hour.

This is probably related to the shader TIME rollover logic set by default to 3600 seconds. If resuming only after 15 minutes for example there is no problem. Tested also with the delta smoothnig setting on and off - same problem in both cases on the said iPad.

### Steps to reproduce

Start the game. Suspend it by exiting the app (via the button or slider). Wait 1 hour. Resume the app.

### Minimal reproduction project

Any project with moving visuals. Particles or shaders using `TIME` are particularly visible.

**Comment by Calinou** [comment]
> This is probably related to the shader TIME rollover logic set by default to 3600 seconds. 

If you increase this value to something like 86400 (24 hours), do you still get the issue after resuming the app after waiting for 1.5 hours or so?

**Comment by lawnjelly** [comment]
Ah it is possible this is due to delta smoothing if this is new, let me take a look. We may need a protection case for long frames.

**Comment by lawnjelly** [comment]
The above PR may well fix it. I can't easily test it of course, but this was a valid bug, and it might well produce the symptoms you describe. Differences on different platforms could be due to whether they use 32bit / 64 CPU, or whether the OS itself does timer capping in such situations.

**Comment by akien-mga** [comment]
Fixed by #51495 (to be confirmed in next beta build).

**Comment by oeleo1** [comment]
Yes, this was the delta smoothing overflow bug. I can confirm that with a shader ˋTIMEˋrollover of 600 (10 minutes) there is no problem on resuming after 20 minutes, no problem after 30 minutes and it goes crazy after 40 minutes.
